src/core/structures/acroform/PDFAcroField.ts

```diff
@@ -55,21 +60,45 @@ class PDFAcroField {
     return this.dict.lookupMaybe(PDFName.of('TM'), PDFString);
   }
 
-  Ff(): PDFNumber | undefined {
-    return this.dict.lookupMaybe(PDFName.of('Ff'), PDFNumber);
+  Ff(): PDFNumber {
+    const ownFieldFlags = this.dict.lookupMaybe(PDFName.Ff, PDFNumber);
+    const fieldFlags =
+      ownFieldFlags || this.getInheritableAttribute(PDFName.Ff, PDFNumber);
+    return fieldFlags || PDFNumber.of(0);
   }
 
   V(): PDFObject | undefined {
-    return this.dict.lookup(PDFName.of('V'));
+    const ownValue = this.dict.lookup(PDFName.V);
+    return ownValue || this.getInheritableAttribute(PDFName.V);
   }
 
   DV(): PDFObject | undefined {
-    return this.dict.lookup(PDFName.of('DV'));
+    const ownDefaultValue = this.dict.lookup(PDFName.DV);
+    return ownDefaultValue || this.getInheritableAttribute(PDFName.DV);
   }
 
   AA(): PDFDict | undefined {
     return this.dict.lookupMaybe(PDFName.of('AA'), PDFDict);
   }
+
+  getInheritableAttribute(name: PDFName): PDFObject | undefined;
+  getInheritableAttribute(name: PDFName, type: typeof PDFName): PDFName;
+  getInheritableAttribute(name: PDFName, type: typeof PDFNumber): PDFNumber;
+
+  getInheritableAttribute(name: PDFName, type?: any): PDFObject | undefined {
+    let attribute: PDFObject | undefined;
+    this.ascend((node) => {
+      if (!(type || attribute)) attribute = node.dict.lookup(name);
+      else if (!attribute) attribute = node.dict.lookupMaybe(name, type);
+    });
+    return attribute;
+  }
```

Comment:
I'm a little concerned that this method forces you to do a lookup. It does not allow you to simply obtain an attribute by reference (if that is how it is defined). I think this is a bit too inflexible. I think it would be better if this method just did a `get` and not a `lookup`. For example:
```js
getInheritableAttribute(name: PDFName): PDFObject | undefined {
  let attribute: PDFObject | undefined;
  this.ascend((node) => {
    if (!attribute) attribute = node.get(name);
  });
  return attribute;
}
```

If you need to do a lookup on the result, that is straightforward:
```js
const objectOrRef = this.getInheritableAttribute(PDFName.Foo);
const object = this.dict.context.lookup(objectOrRef, PDFArray);
```

But if this boilerplate is too excessive, we could introduce a `lookupInheritableAttribute` method as well:
```js
lookupInheritableAttribute(name: PDFName, type?: any): PDFObject | undefined {
  const objectOrRef = this.getInheritableAttribute(PDFName.Foo);
  return this.dict.context.lookup(objectOrRef, type);
}
```

Reply:
Updated